### PR TITLE
Network visitor 

### PR DESCRIPTION
**Pull request by petersteneteg**
rebase of #947

### CODE REVIEW

**Comment by petersteneteg** [comment]
@ResearchDaniel rebased this one...

File: include/inviwo/core/network/lambdanetworkvisitor.h [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        } else {
+            return true;
+        }
+    }
+
+    virtual bool visit([[maybe_unused]] Property& property) override {
+        if constexpr (util::is_detected_exact_v<bool, PropertyOverload, decltype(*this)>) {
+            return operator()(property);
+        } else {
+            return true;
```
Comment (ResearchDaniel) [inline include/inviwo/core/network/lambdanetworkvisitor.h:97]: Should be false?

File: include/inviwo/core/properties/propertyowner.h [hunk 2]
```diff
@@ -153,39 +154,31 @@ class IVW_CORE_API PropertyOwner : public PropertyOwnerObservable,
     // allocated on the heap.
     std::vector<std::unique_ptr<Property>> ownedProperties_;
 
+    void forEachProperty(std::function<void(Property&)> callback,
+                         bool recursiveSearch = false) const;
```
Comment (ResearchDaniel) [inline include/inviwo/core/properties/propertyowner.h:158]: Const?

**Review by ResearchDaniel — APPROVED** [review]
Check the comments

File: include/inviwo/core/properties/propertyowner.h (hunk 2 above)
Comment (petersteneteg) [reply include/inviwo/core/properties/propertyowner.h:158]: all the other property getter are const...

File: include/inviwo/core/network/lambdanetworkvisitor.h (hunk 1 above)
Comment (petersteneteg) [reply include/inviwo/core/network/lambdanetworkvisitor.h:97]: we generally want to do a recursive traversal which is what the parameter signals, although for a property that is not used, it's only there to be consistent with the other ones...

**Pull request merged**